### ISSUE TITLE
Forward slash in strings results in incorrect message

### ISSUE THREAD

**Issue by stevenh** [CodeNarc/CodeNarc#749]
When processing a file which includes double quotes that could be single and the value includes a backslash the resulting output in message has the wrong value.

Test file
```groovy
/**
 * Wibble
 */
class Wibble {

    public static final String DOUBLE = "C:\\wibble"

    static printBlip() {
        println 'blip'
    }

}
```

```json
"violations": [
  {
    "ruleName": "UnnecessaryGString",
    "priority": 3,
    "lineNumber": 6,
    "sourceLine": "public static final String DOUBLE = \"C:\\\\wibble\"",
    "message": "The String 'C:\\wibble' can be wrapped in single quotes instead of double quotes"
  }
]
```

You can see that the `sourceLine` is correct including four backslashes but the message has had them stripped and only contains two.

**Comment by stevenh** [comment]
Correct, otherwise when used to display an error to the user the `String` is not actually present in the file as it will display:
`The String 'C:\wibble' can be wrapped in single quotes instead of double quotes`
instead of:
`The String 'C:\\wibble' can be wrapped in single quotes instead of double quotes`

**Comment by chrismair** [comment]
Will be available in CodeNarc 3.4.0

**Comment by stevenh** [comment]
thanks for the heads up @chrismair, PR is up to revert: https://github.com/nvuillam/npm-groovy-lint/pull/545
